tests/integ/modin/index/test_sort_values.py

```diff
@@ -0,0 +1,215 @@
+    (
+        native_pd.Index(
+            [1.1, np.nan, 2.2, 1.0, 1, 1.1, 2.2, 1, np.nan, 1, 1, 2, 2, 2, np.nan, 2.2]
+        ),
+        True,
+    ),
+    (native_pd.Index([1, 3, 1, 1, 1, 3, 1, 1, 1, 2, 2, 2, 3]), True),
+    (
+        native_pd.Index(
+            [True, False, True, False, True, False, True, False, True, True], dtype=bool
+        ),
+        True,
+    ),
+    (
+        native_pd.Index(
+            [
+                "a",
+                "b",
+                "c",
+                "b",
+                "c",
+                "e",
+                "f",
+                "g",
+                "h",
+                "d",
+                "d",
+                "e",
+                "f",
+                "g",
+                "a",
+                "b",
+                "a",
+                "b",
+                "c",
+                "d",
+                "c",
+                "d",
+                "e",
+                "f",
+                "g",
+            ]
+        ),
+        False,
+    ),
+]
+
+
+@pytest.mark.parametrize("native_index", SINGLE_TYPE_NATIVE_INDEX_TEST_DATA)
+@pytest.mark.parametrize("ascending", [True, False])
+@pytest.mark.parametrize("na_position", ["first", "last"])
+@sql_count_checker(query_count=1)
+def test_index_sort_values(native_index, ascending, na_position):
+    # In this test, `return_indexer` is False: only an Index is returned.
+    snow_index = pd.Index(native_index)
+    snow_res = snow_index.sort_values(
+        return_indexer=False, ascending=ascending, na_position=na_position
+    )
+    native_res = native_index.sort_values(
+        return_indexer=False, ascending=ascending, na_position=na_position
+    )
+    assert_index_equal(snow_res, native_res)
+
+
+@pytest.mark.parametrize(
+    "native_index, match_pandas", SINGLE_TYPE_NATIVE_INDEX_TEST_DATA_WITH_INDEXER_MATCH
+)
+@pytest.mark.parametrize("ascending", [True, False])
+@pytest.mark.parametrize("na_position", ["first", "last"])
+@sql_count_checker(query_count=2)
+def test_index_sort_values_return_indexer(
+    native_index, match_pandas, ascending, na_position
+):
+    # In this test, `return_indexer` is True: a tuple of the result Index
+    # and a numpy ndarray of the indexer is returned.
+    snow_index = pd.Index(native_index)
+    snow_res = snow_index.sort_values(
+        return_indexer=True, ascending=ascending, na_position=na_position
+    )
+    native_res = native_index.sort_values(
+        return_indexer=True, ascending=ascending, na_position=na_position
+    )
+
+    # Compare the index part of the results.
+    assert_index_equal(snow_res[0], native_res[0])
+
+    # Pandas performs quicksort to generate the indexer. Currently, Snowpark pandas does not support quicksort;
+    # instead, stable sort is performed. Therefore, the results are not guaranteed to match.
+    # In case the Snowpark pandas result does not match pandas, verify that it matches `np.argsort(kind=stable)`.
+    if match_pandas:
```

Comment:
match_pandas is True for the indices without duplicates?
Given pandas output is non -stable if index has duplicates we should consider doing np.argesort(kind=stable) whenever duplicate values are present.

Reply:
No, all the example tests have duplicates in them. It's just that when `match_pandas=True`, quicksort and stable sort produce the same output/indexer.

Reply:
I see. One potential issue we have that test can become flaky when output of quicksort changes (due to non-stable nature).

Reply:
If all have duplicate values. I think we should just use what you have in else block to compare results.

Reply:
okay, I'll change it to compare with only numpy stable argsort